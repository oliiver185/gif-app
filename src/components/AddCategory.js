import React, { useState } from 'react';
import PropTypes from 'prop-types'



const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');  //useState() asi es undefined
    
    const handleInputChange = (e) => {
        setInputValue (e.target.value);
    }

    const handleSubmit = (e) => {
          e.preventDefault(); 
          

        if (inputValue.trim().length > 2){
          setCategories(cats => [inputValue, ...cats ]);
          setInputValue('');
        }    

        // inputValue.trim().length > 2 && setCategories(cats => [...cats, inputValue]);
        // inputValue && setCategories( (category) => [...category, inputValue] ); 
    }

    return (       
        <form onSubmit={handleSubmit}>
          <p>{inputValue}</p>
        <input
          type="text"
          value = {inputValue}
          onChange= { handleInputChange}
          placeholder="Type something ðŸ˜€"
          /> 
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

// PrimeraApp.propTypes = {
//   saludo: PropTypes.string.isRequired
// //    otra: PropTypes.number.isRequired 
// }
export default AddCategory;