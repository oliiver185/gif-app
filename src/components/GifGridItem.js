import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({title, url}) => {
    
    // console.log({id, title, url})
    return (
        <div className="card animate__animated animate__fadeIn">
           <img src={url} alt={title}  />

           

           <div className="hover-galeria">
               <a href={url}><p>{title}</p></a>
           </div>
           
        </div>
    )
}


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}


/*

1.enzyne
2. enzyme to json
3. deve de mostrar el componente correctamente 

*shallow
*wrapper
*wrapper .tomatchSnapshot()


*/