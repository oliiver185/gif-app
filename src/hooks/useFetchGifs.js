import { useEffect, useState } from "react"
import {getGif} from '../helpers/getGifs'

export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data:[],
        loading: true
    });


   useEffect(() => {
    
    getGif( category).then(img => {
        
        // setTimeout(() => {

            // console.log(img)
            setState({
                data: img,
                loading: false
            })
        // }, 3000);
        
        
        
    })

   },[category])

    // setTimeout(() => {
    //     setState({
    //         data:[1,2,3,4,5],
    //         loading: false
    //     })
    // }, 3000)

    return state;
}
