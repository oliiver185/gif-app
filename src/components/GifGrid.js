import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGif } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

      const {data: images,  loading } = useFetchGifs(category);
     
      
    // const [count, setCount] = useState(0);

    // const [images, setImages] = useState([])

    

   

    // getGif();




    return (
<>
        <h3 className="animate__animated animate__fadeIn third-title">{category}</h3>

        {loading && <p className="animate__animated animate__flash">Loading...</p>}

        {/* {loading ? 'cargando' : 'data cargada'}  */}
        <div className="card-grid">
            
            

            {/* en esta parte se puede hacer la desestructuracion de un vez sin necesidad de poner un call
            back como por ejemplo map(imagen => <li key{imagen.id>{imagen.title}</li */}

             {
               images.map(img => (
                   <GifGridItem 
                   key ={img.id}
                    {...img} /> 
                   ))
                   
            }
          
          
        </div> 

        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}