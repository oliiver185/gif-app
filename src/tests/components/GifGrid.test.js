import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');



const category = 'hola mundo'

describe('Working with <GifGrid />', () => {

    test('should should work correctly', () => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true  
        });


        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();

    })
    

    test('should show items when it is loading images from useFetchGifs ', () => {
        

       const gifs = [{
           id: 'ABC',
           url: 'htpps://algo.jpg',
           title: 'cualquier cosa'
       }] 
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false  
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    }); 
    
    
})
