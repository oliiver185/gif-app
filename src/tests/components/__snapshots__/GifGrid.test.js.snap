// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Working with <GifGrid /> should should work correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn third-title"
  >
    hola mundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Working with <GifGrid /> should show items when it is loading images from useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn third-title"
  >
    hola mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="htpps://algo.jpg"
    />
  </div>
</Fragment>
`;
